### PR TITLE
Fixes #514 SwapWidget changes from transaction submitted to transaction completed

### LINKED ISSUE #514: If "Transaction Submitted" screen is still visible when transaction completes, update the text

### PR DESCRIPTION
**Pull request by ilam0602**
Fixes #514 ;)

### CODE REVIEW

File: public/locales/fr/translation.json [hunk 1]
```diff
@@ -104,6 +104,7 @@
         "swapMessage": "Acceptez sur votre wallet pour continuer",
         "wrapMessage": "Pas de frais de protocole pour les swaps ETH/WETH",
         "submitted": "Transaction soumise",
+        "transactionCompleted": "Transaction terminée",
```
Comment (piersss) [inline public/locales/fr/translation.json:107]: These translation.json files will be overwritten by our translation too. You only have to do the english one for now.

File: src/contexts/interface/Interface.tsx [hunk 2]
```diff
@@ -11,13 +11,15 @@ export interface InterfaceContextContextProps {
   showMobileToolbar: boolean;
   showWalletList: boolean;
   transactionsTabIsOpen: boolean;
+  transactionComplete: boolean;
```
Comment (piersss) [inline src/contexts/interface/Interface.tsx:14]: Interface is only for things related to interface, it's not for individual transactions.

File: src/components/TransactionsTab/TransactionsTab.tsx [hunk 3]
```diff
@@ -150,6 +153,14 @@ const TransactionsTab = ({
...
   const balance = balances.values[nativeCurrencyAddress] || "0";
 
+  useEffect(() => {
+    if (completedTransactions.length > 1 && pendingTransactions.length === 0) {
+      setTransactionComplete(true);
+    } else {
+      setTransactionComplete(false);
+    }
+  }, [completedTransactions, pendingTransactions, setTransactionComplete]);
+
```
Comment (piersss) [inline src/components/TransactionsTab/TransactionsTab.tsx:163]: You should write this kind of logic in the transactions store and not in a component. You should also look for a specific transaction, you should have a nonce when you submit a transaction, just wait for it to complete and your transaction is complete.

**Review by piersss — CHANGES_REQUESTED** [review]
Please tag your ticket like so: Fixes #514 

Also drag your ticket to "in progress" when you're working on it.

How did you test this? Any working pairs on goerli? Or are you on different chain?

File: src/components/TransactionsTab/TransactionsTab.tsx (hunk 3 above)
Comment (piersss) [reply src/components/TransactionsTab/TransactionsTab.tsx:163]: I think you should just make a hook, look at `useCancellationSuccess` for inspiration.

**Comment by ilam0602** [comment]
I tested with eth/weth on goerli

File: public/locales/fr/translation.json (hunk 1 above)
Comment (ilam0602) [reply public/locales/fr/translation.json:107]: Okay sounds good I was unsure whether or not I had to change the other translations

File: src/components/SwapWidget/subcomponents/InfoSection/InfoSection.tsx [hunk 4]
```diff
@@ -33,7 +35,7 @@ export type InfoSectionProps = {
         pricing: Levels;
       }
     | {
-        protocol: "request-for-quote";
+        protocol: "request-for-quote"
```
Comment (ilam0602) [inline src/components/SwapWidget/subcomponents/InfoSection/InfoSection.tsx:38]: i accidentally deleted the semi colon i will make sure not to merge this

File: src/components/SwapWidget/SwapWidget.tsx [hunk 5]
```diff
@@ -192,6 +195,12 @@ const SwapWidget: FC = () => {
     baseTokenInfo?.address === nativeCurrencyAddress &&
     !!nativeCurrencySafeTransactionFee[baseTokenInfo.chainId];
 
+  useEffect(()=>{
+    if(showOrderSubmitted&&currOrder){
+      setOrderNonce((currOrder!.nonce!));
```
Comment (ilam0602) [inline src/components/SwapWidget/SwapWidget.tsx:200]: this was the way i was able to get the nonce. i tried getting it from besTradeOption.order.nonce but that was not working well for me. is this a good way to get the nonce?

**Comment by piersss** [comment]
@ilam0602 Please remove all unnecessary code from your PR first. The translation files, yarn.lock changes, empty enters.

**Comment by ilam0602** [comment]
@piersss done!

File: src/components/SwapWidget/SwapWidget.tsx [hunk 6]
```diff
@@ -113,6 +114,7 @@ const SwapWidget: FC = () => {
   const supportedTokens = useAppSelector(selectAllSupportedTokens);
   const tradeTerms = useAppSelector(selectTradeTerms);
   const userTokens = useAppSelector(selectUserTokens);
+  const currOrder = useAppSelector(selectPendingTransactions)[0];
```
Comment (piersss) [inline src/components/SwapWidget/SwapWidget.tsx:117]:
```suggestion
  const pendingTransactions = useAppSelector(selectPendingTransactions);
```

File: src/components/SwapWidget/SwapWidget.tsx [hunk 7]
```diff
@@ -192,6 +195,12 @@ const SwapWidget: FC = () => {
     baseTokenInfo?.address === nativeCurrencyAddress &&
     !!nativeCurrencySafeTransactionFee[baseTokenInfo.chainId];
 
+  useEffect(() => {
+    if (showOrderSubmitted && currOrder) {
+      setOrderNonce(currOrder!.nonce!);
+    }
+  }, [showOrderSubmitted, currOrder]);
```
Comment (piersss) [inline src/components/SwapWidget/SwapWidget.tsx:202]:
```  
useEffect(() => {
    if (showOrderSubmitted && pendingTransactions[0]?.nonce) {
      setOrderNonce(pendingTransactions[0].nonce);
    }
  }, [showOrderSubmitted, pendingTransactions]);
```

File: src/components/SwapWidget/SwapWidget.tsx [hunk 8]
```diff
@@ -143,6 +145,7 @@ const SwapWidget: FC = () => {
     useState<TokenSelectModalTypes | null>(null);
   const [showGasFeeInfo, setShowGasFeeInfo] = useState(false);
   const [protocolFeeDiscountInfo, setProtocolFeeDiscountInfo] = useState(false);
+  const [orderNonce, setOrderNonce] = useState<string>("");
```
Comment (piersss) [inline src/components/SwapWidget/SwapWidget.tsx:148]:
```suggestion
  const [lastSubmittedOrderNonce, setLastSubmittedOrderNonce] = useState<string | undefined>();
```

**Review by piersss — CHANGES_REQUESTED** [review]
I think this is the right way yeah. It doesn't work properly though, it defaults to "Transaction Completed" after submitting.

Can you not put the `usePendingTransaction` in the InfoSection and keep the hooks in `SwapWidget`. Right now `InfoSection` has a `orderSubmitted` prop. Split it into `orderSubmitPending` and `orderCompleted`.

Also make sure to do some cleanup in `case ButtonActions.restart:` line 604.

**Comment by ilam0602** [comment]
Okay I moved the hook out and made suggested changes. Regarding the comment about how it defaults to "Transaction Completed" after submitting, since orderNonce gets reset to undefined when `case ButtonActions.restart:` occurs, this is now default to "Transaction Submitted". Is this correct? If not do you want me to check that the nonce appears in orderCompleted rather than seeing when it is no longer in orderProcessing.

**Comment by piersss** [comment]
@ilam0602 Actually things can be made much simpler so I removed some code. If we can assume that the last transaction from `selectTransactions` is the one we need then we need a lot less variables.

**Review by piersss — APPROVED** [review]

**Pull request merged**